MapboxNavigation/NavigationViewController.swift

```diff
@@ -21,6 +21,14 @@ public protocol NavigationViewControllerDelegate {
      Called when the user arrives at the destination.
      */
     @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt destination: MGLAnnotation)
+
+    /**
+     Called when `RouteControllerShouldReroute` is sent from `RouteController`.
+
+     If this method is unimplemented `NavigationViewController` will automatically reroute. If this method is implemented,
+     it's responsible for handling rerouting.
+    */
+    @objc optional func navigationViewControllerShouldReroute(_ : NavigationViewController, fromLocation location: CLLocation)
```

Comment:
The method should be called `navigationViewController(_:shouldRerouteFrom:)`, with an attribute of `@objc(navigationViewController:shouldRerouteFromLocation:)`.